test/blas_test.hpp

```diff
@@ -204,8 +204,13 @@ class BLAS_Test<blas_test_args<ScalarT_, MetadataT_, ExecutorType_>>
   }
 };
 
-#define ASSERT_T_EQUAL(T, val1, val2)                                         \
-  ASSERT_PRED_FORMAT2(::testing::internal::CmpHelperFloatingPointEQ<T>, val1, \
-                      val2)
+constexpr const int ulp_required_in_assert = 20;
+
+#define ASSERT_T_EQUAL(T, val1, val2)                                   \
```

Comment:
create a function called precision Computation and return the precision. So in this case you dont need macro for asser_near and you can directly put it in the code

Reply:
Should be done now.